src/rust/engine/fs/store/src/local.rs

```diff
@@ -395,26 +448,36 @@ impl ByteStore {
       return Ok(Some(f(&[])));
     }
 
-    let dbs = match entry_type {
-      EntryType::Directory => self.inner.directory_dbs.clone(),
-      EntryType::File => self.inner.file_dbs.clone(),
-    }?;
-    let res = dbs
-      .load_bytes_with(digest.hash, move |bytes| {
-        if bytes.len() == digest.size_bytes {
-          Ok(f(bytes))
-        } else {
-          Err(format!(
-            "Got hash collision reading from store - digest {:?} was requested, but retrieved \
-                bytes with that fingerprint had length {}. Congratulations, you may have broken \
-                sha256! Underlying bytes: {:?}",
-            digest,
-            bytes.len(),
-            bytes
-          ))
-        }
-      })
-      .await;
+    let res = if ByteStore::uses_large_file_store(digest.size_bytes) {
+      let src = &self.large_file_path(digest);
+      let mut reader = std::fs::File::open(src)
```

Comment:
Is this an `async` hazard?  Maybe it could either:

- be using `tokio::fs::File` to be able to do async reads? 
- shunt the file reading out of the executor thread via `spawn_blocking`?

Or is this not a problem for file reading in pants?

(The same question applies to callers of the new functions in `hashing` like `hash_path` and `verified_copy`: they're doing sync IO, but called in async contexts.)

Reply:
I had this thought as well, but the old code used this API so I kept it this way 😬

Reply:
Hm, maybe I'm misreading it, but I think the old code used only touched `std::fs::File` within a closure passed to `ShardedLmdb::store`, and all the interesting code there (including calling that closure/manipulating the resulting `std::fs::File`) is wrapped in `self.execute.spawn_blocking(...)`, whereas this seems to be doing `std::fs::File` IO directly within `async` code?

Reply:
Ahh ok yeah that makes sense. I'll fix this when I'm back

Reply:
Yea, this PR should use `tokio::fs::File` ~everywhere, although unless you hit a deadlock, you should be fine to wait until all other issues are resolved before find-replacing.

In this _particular_ function though, we might actually eventually be able to use `mmap` rather than copying the file into user space. Better as a TODO and followup though.